pkg/controllers/static/provisioning/controller.go

```diff
@@ -0,0 +1,149 @@
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	nodeutils "sigs.k8s.io/karpenter/pkg/utils/node"
+
+	"sigs.k8s.io/karpenter/pkg/controllers/provisioning"
+	"sigs.k8s.io/karpenter/pkg/controllers/state"
+	"sigs.k8s.io/karpenter/pkg/events"
+	"sigs.k8s.io/karpenter/pkg/metrics"
+
+	"sigs.k8s.io/karpenter/pkg/cloudprovider"
+
+	v1 "sigs.k8s.io/karpenter/pkg/apis/v1"
+	helper "sigs.k8s.io/karpenter/pkg/controllers/static"
+	"sigs.k8s.io/karpenter/pkg/operator/injection"
+	nodepoolutils "sigs.k8s.io/karpenter/pkg/utils/nodepool"
+)
+
+type Controller struct {
+	kubeClient    client.Client
+	cloudProvider cloudprovider.CloudProvider
+	provisioner   *provisioning.Provisioner
+	cluster       *state.Cluster
+	recorder      events.Recorder
+}
+
+func NewController(kubeClient client.Client, cluster *state.Cluster, recorder events.Recorder, cloudProvider cloudprovider.CloudProvider, provisioner *provisioning.Provisioner, clock clock.Clock) *Controller {
+	return &Controller{
+		kubeClient:    kubeClient,
+		cloudProvider: cloudProvider,
+		cluster:       cluster,
+		recorder:      recorder,
+		provisioner:   provisioning.NewProvisioner(kubeClient, recorder, cloudProvider, cluster, clock),
+	}
+}
+
+// Reconcile the resource
+func (c *Controller) Reconcile(ctx context.Context, np *v1.NodePool) (reconcile.Result, error) {
+	ctx = injection.WithControllerName(ctx, "provisioning.static")
+
+	if !nodepoolutils.IsManaged(np, c.cloudProvider) || !np.StatusConditions().Root().IsTrue() || np.Spec.Replicas == nil {
+		return reconcile.Result{}, nil
+	}
+
+	nodes := helper.TotalNodesForNodePool(c.cluster, np)
+	// Size down of replicas will be handled in disruption controller to drain nodes and delete NodeClaims
+	if nodes >= lo.FromPtr(np.Spec.Replicas) {
+		return reconcile.Result{}, nil
+	}
+
+	countNodeClaimsToProvision := helper.ComputeNodeClaimsToProvision(c.cluster, np, nodes)
+	if countNodeClaimsToProvision <= 0 {
+		log.FromContext(ctx).WithValues("NodePool", klog.KObj(np)).Info("nodepool node limit reached")
+		return reconcile.Result{RequeueAfter: time.Second * 30}, nil
+	}
+
+	its, err := c.cloudProvider.GetInstanceTypes(ctx, np)
+	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			return reconcile.Result{}, fmt.Errorf("timed out while getting instance types: %w", err)
+		}
+		log.FromContext(ctx).WithValues("NodePool", klog.KObj(np)).Error(err, "failed to resolve instance types")
+		return reconcile.Result{}, fmt.Errorf("failed to resolve instance types: %w", err)
+	}
+
+	nodeClaims := helper.GetStaticNodeClaimsToProvision(np, its, countNodeClaimsToProvision)
+
+	_, err = c.provisioner.CreateNodeClaims(ctx, nodeClaims, provisioning.WithReason(metrics.ProvisionedReason))
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("creating nodeclaims, %w", err)
+	}
+
```

Comment:
Should we add a log line here along the lines of "provisioned %d nodeclaims" with the nodepool as an attached value? You probably wouldn't want to use `CreateNodeClaims` directly since it wouldn't let you check how many you created in a partial success case, but you could still reuse `Create` from the provisioner.